# vsearchpipeline: Parameters

| Group                   | Property                    | Type      | Description                                                  | Default Value | Required | Fa Icon                  | Help Text                                                                                                                                                                                                                        |
|-------------------------|-----------------------------|-----------|--------------------------------------------------------------|---------------|----------|--------------------------|----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| Input/output options    | input                       | string    | Path to comma-separated file containing information about the samples in the experiment. | -             | *        | fas fa-file-csv           | You will need to create a design file with information about the samples in your experiment before running the pipeline. Use this parameter to specify its location. It has to be a comma-separated file with 3 columns, and a header row. |
|                         | primers                     | string    | Path to comma-separated file containing forward_primer and reverse_primer sequences. | -             |          | fas fa-file-csv           |                                                                                                                                                                                                                                  |
|                         | outdir                      | string    | The output directory where the results will be saved. You have to use absolute paths to storage on Cloud infrastructure. | -             | *        | fas fa-folder-open        |                                                                                                                                                                                                                                  |
|                         | email                       | string    | Email address for completion summary.                        | -             |          | fas fa-envelope           | Set this parameter to your e-mail address to get a summary e-mail with details of the run sent to you when the workflow exits. If set in your user config file (`~/.nextflow/config`) then you don't need to specify this on the command line for every run. |
|                         | multiqc_title               | string    | MultiQC report title. Printed as page header, used for filename if not otherwise specified. | -             |          | fas fa-file-signature     |                                                                                                                                                                                                                                  |
| VSEARCH options          | merge_allowmergestagger     | boolean   | Fastq merge process: allow merging of staggered read pairs.   | -             |          | fas fa-hashtag            | When using --fastq_mergepairs, allow merging of staggered read pairs. Staggered pairs are pairs where the 3’ end of the reverse read has an overhang to the left of the 5’ end of the forward read. This situation can occur when a very short fragment is sequenced. The 3’ overhang of the reverse read is not included in the merged sequence. The opposite option is the --fastq_nostagger option. The default is to discard staggered pairs. |
|                         | merge_maxdiffs              | integer   | Fastq merge process: Discard input sequences with more bases difference than specified | 30            |          | fas fa-exchange-alt       | When using --fastq_mergepairs, specify the maximum number of non-matching nucleotides allowed in the overlap region. That option has a strong influence on the merging success rate. The default value is 10.                                                                             |
|                         | merge_minlen                | integer   | Fastq merge process: Discard input sequences shorter than the specified number of bases | 0             |          | fas fa-ruler-horizontal   |                                                                                                                                                                                                                                  |
|                         | merge_maxdiffpct            | integer   | Fastq merge process: maximum percentage of non-matching nucleotides allowed in the overlap region | 100           |          | fas fa-ruler-horizontal   |                                                                                                                                                                                                                                  |
|                         | merge_maxlen                | integer   | Fastq merge process: Discard sequences longer than the specified number of bases | 0             |          | fas fa-ruler-vertical     |                                                                                                                                                                                                                                  |
|                         | filter_maxee                | number    | Filtering process: Discard sequences with an expected error greater than the specified number | 1             |          | fas fa-times              |                                                                                                                                                                                                                                  |
|                         | filter_maxns                | integer   | Filtering process: Discard sequences with more than the specified number of N’s. | 0             |          | fas fa-sort-numeric-down  |                                                                                                                                                                                                                                  |
|                         | filter_minlen               | integer   | Filtering process: Discard sequences shorter than the specified length. | 0             |          | fas fa-sort-numeric-down  |                                                                                                                                                                                                                                  |
|                         | filter_maxlen               | integer   | Filtering process: Discard sequences longer than the specified length. | 0             |          | fas fa-sort-numeric-down  |                                                                                                                                                                                                                                  |
|                         | derep_strand                | string    | Dereplicate process: plus or both strands                     | plus          |          | fas fa-plus               |                                                                                                                                                                                                                                  |
|                         | derep_all_strand            | string    | Dereplicate process all samples: plus or both strands         | plus          |          | fas fa-plus               |                                                                                                                                                                                                                                  |
|                         | derep_all_fastawidth        | integer   | Dereplicate process all samples:                              | 0             |          | fas fa-arrows-alt-h       |                                                                                                                                                                                                                                  |
|                         | derep_all_minunique         | integer   | Dereplicate process all samples                               | 2             |          | fas fa-fingerprint        |                                                                                                                                                                                                                                  |
|                         | cluster_minsize             | integer   | Clustering                                                   | 8             |          | fas fa-compress-alt       |                                                                                                                                                                                                                                  |
|                         | cluster_alpha               | number    | Clustering                                                   | 2.0           |          | fas fa-font               |                                                                                                                                                                                                                                  |
|                         | sort_fastawidth             | integer   | Sorting and removing singletons: fastawidth                   | 0             |          | fas fa-compress-alt       |                                                                                                                                                                                                                                  |
|                         | sort_minsize                | integer   | Sorting and removing singletons: minsize                      | 2             |          | fas fa-compress-alt       |                                                                                                                                                                                                                                  |
|                         | uchime_label                | string    | Chimera removal: labeling (prefix) of ASVs                    | ASV_          |          | fas fa-tag                |                                                                                                                                                                                                                                  |
|                         | usearch_id                  | number    | Usearch global: id parameter                                  | 0.97          |          | far fa-address-card       |                                                                                                                                                                                                                                  |
| DADA2 options            | dada2_minboot               | integer   | assignTaxonomy function: The minimum bootstrap confidence for assigning a taxonomic level. | 80            |          | fas fa-terminal           |                                                                                                                                                                                                                                  |
|                         | dada2_allowmultiple         | integer   | addSpecies function: maximum number of multiple assigned species. If 0, this will be set at FALSE. | 3             |          | fas fa-cubes              |                                                                                                                                                                                                                                  |
|                         | dada2_tryrevcompl           | boolean   | addSpecies function: If TRUE, the reverse-complement of each sequences will be used for classification if it is a better match to the reference sequences than the forward sequence. | true          |          | fas fa-arrow-left         |                                                                                                                                                                                                                                  |
| Phyloseq options         | rarelevel                   | integer   | Rarefaction level (not used if skip_rarefaction is not set at true) | 0             |          | fas fa-th-list            |                                                                                                                                                                                                                                  |
| Skip options             | skip_primers                | boolean   | Skip trimming of primers                                    | -             |          | fas fa-forward            |                                                                                                                                                                                                                                  |
|                         | skip_tree                   | boolean   | Skip multiple sequence alignment and tree processes           | -             |          | fas fa-forward            |                                                                                                                                                                                                                                  |
|                         | skip_rarefaction            | boolean   | Skip rarefaction of phyloseq object                           | -             |          | fas fa-forward            |                                                                                                                                                                                                                                  |
|                         | skip_fixtaxonomy            | boolean   | Skip process to make a table with composite taxonomy names    | -             |          | fas fa-forward            |                                                                                                                                                                                                                                  |
|                         | skip_metrics                | boolean   | Skip process to provide some general metrics of the phyloseq object (composition, diversity) | -             |          | fas fa-forward            |                                                                                                                                                                                                                                  |
